HW02/Alexey_Zalyotov/hw_02_t_03/lib/log_formatter.rb

```diff
@@ -0,0 +1,27 @@
+require 'date'
+
+class LogFormatter
+  EVENT_REGEX = /.*Calling core.*/.freeze
+  DATE_TIME_REGEX = /^[[:alnum:]\.\:\-\s]{21}/.freeze
+
+  attr_reader :logs
+
+  def initialize(logs)
+    @logs = logs
+  end
+
+  def events_timestamps
+    # find lines with "Calling core with action" substr
+    # and parse their timestamps
+    logs.select { |line| line =~ EVENT_REGEX }.map do |substr|
+      substr[DATE_TIME_REGEX]
```

Comment:
Maybe we don't need `datetimestamps` method at all? Why don't you want to do it by this way?
```suggestion
      DateTime.parse(substr[DATE_TIME_REGEX])
```

Reply:
Because I used `log_formatter.events_timestamps` to make the formatted output as assigned in the task:

> For this example, the output will be: (2018-04-23 17:18:38.8 - 2018-04-23 17:17:49.7)...

Reply:
You can transform a real date to this format with https://ruby-doc.org/stdlib-2.6.1/libdoc/date/rdoc/DateTime.html#method-i-strftime, but I think we just need to return a string with a number  (`"49.1"`)

Reply:
Normally, one class should have only one main public method ( single responsibility principle )

Reply:
> single responsibility principle
Right, **S**OLID )